src/controller/java/OTAProviderDelegateBridge.cpp

```diff
@@ -0,0 +1,424 @@
+
+    jobject otaProviderDelegate = mOtaProviderDelegate.ObjectRef();
+    jmethodID handleOTAQueryFailureMethod;
+    CHIP_ERROR err = JniReferences::GetInstance().FindMethod(env, otaProviderDelegate, "handleOTAQueryFailure", "(I)V",
+                                                             &handleOTAQueryFailureMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find handleOTAQueryFailure method"));
+
+    env->CallVoidMethod(otaProviderDelegate, handleOTAQueryFailureMethod, static_cast<jint>(status));
+    if (env->ExceptionCheck())
+    {
+        ChipLogError(Support, "Exception in call java method");
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+        return;
+    }
+    return;
+}
+void OTAProviderDelegateBridge::HandleQueryImage(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                 const QueryImage::DecodableType & commandData)
+{
+    CHIP_ERROR err      = CHIP_NO_ERROR;
+
+    NodeId nodeId = kUndefinedNodeId;
+
+    FabricIndex fabricIndex = kUndefinedFabricIndex;
+    ScopedNodeId ourNodeId;
+
+    VendorId vendorId        = VendorId::Unspecified;
+    uint16_t productId       = 0;
+    uint32_t softwareVersion = 0;
+    DataModel::DecodableList<OTADownloadProtocol> protocolsSupported;
+    Optional<uint16_t> hardwareVersion;
+    Optional<chip::CharSpan> location;
+    Optional<bool> requestorCanConsent;
+    Optional<chip::ByteSpan> metadataForProvider;
+
+    bool isBDXProtocolSupported = false;
+
+    ConcreteCommandPath cachedCommandPath(commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    assertChipStackLockedByCurrentThread();
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    jobject otaProviderDelegate = nullptr;
+    chip::JniLocalReferenceManager manager(env);
+    jmethodID handleQueryImageMethod = nullptr;
+
+    jobject boxedHardwareVersion     = nullptr;
+    jobject boxedLocation            = nullptr;
+    jobject boxedRequestorCanConsent = nullptr;
+    jobject boxedMetadataForProvider = nullptr;
+    jobject jResponse                = nullptr;
+
+    jmethodID getSoftwareVersionMethod       = nullptr;
+    jmethodID getSoftwareVersionStringMethod = nullptr;
+    jmethodID getFilePathMethod              = nullptr;
+
+    jobject jSoftwareVersion       = nullptr;
+    jstring jSoftwareVersionString = nullptr;
+    jstring jFilePath              = nullptr;
+
+    bool hasUpdate = false;
+    Commands::QueryImageResponse::Type response;
+    Commands::QueryImageResponse::Type errorResponse;
+    errorResponse.status = OTAQueryStatus::kNotAvailable;
+
+    char uriBuffer[kMaxBDXURILen];
+    MutableCharSpan uri(uriBuffer);
+
+    VerifyOrExit(mOtaProviderDelegate.HasValidObjectRef(),
+                 commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::Failure));
```

Comment:
you add status here, then in exit from Line 322, you add response again?

Reply:
I know that this code is only the status changing and does not send a response.

Reply:
either you add status, commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::Failure));
or 
commandObj->AddResponse(cachedCommandPath, errorResponse);

Reply:
Overall, it has been modified so that either AddStatus or Addresponse is called.